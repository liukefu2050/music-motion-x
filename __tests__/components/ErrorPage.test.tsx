import { fireEvent, render, screen } from "@testing-library/react"
import ErrorPage from "../../src/components/ErrorPage"

it("worked when error", () => {
  const resetErrorBoundary = jest.fn()
  const original = window.console.log
  window.console.log = jest.fn()
  const { asFragment } = render(
    <ErrorPage
      resetErrorBoundary={resetErrorBoundary}
      error={new Error("test")}
    />,
  )

  fireEvent.click(screen.getByTestId("error-btn"))
  expect(window.console.log).toBeCalledTimes(1)
  expect(resetErrorBoundary).toHaveBeenCalledTimes(1)
  expect(asFragment()).toMatchSnapshot()
  window.console.log = original
})
