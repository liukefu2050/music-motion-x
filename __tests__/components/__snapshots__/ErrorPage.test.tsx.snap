// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`worked when error 1`] = `
<DocumentFragment>
  <div
    class=" text-fg  text-lg mt-[35vh] font-bold text-center"
  >
    <div>
      <span>
        Oops! There was an error!
      </span>
       
      <span
        class=" ml-1"
      >
        :(
      </span>
    </div>
    <div
      class=" text-secondary mt-1"
      data-testid="error-btn"
    >
      Try again
    </div>
  </div>
</DocumentFragment>
`;
